libr/anal/p/anal_xtensa.c

```diff
@@ -624,6 +1919,13 @@ static int xtensa_op (RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *buf, int l
 		return 1;
 
 	xtensa_op0_fns[(buf[0] & 0xf)] (anal, op, addr, buf);
+
+	if (anal->decode) {
+		uint8_t copy[XTENSA_MAX_LENGTH];
+		memcpy (copy, buf, op->size);
```

Comment:
you can use R_MIN() for this
> On 01 Sep 2016, at 11:56, Vlad Ivanov <notifications@github.com> wrote:
> 
> In libr/anal/p/anal_xtensa.c <https://github.com/radare/radare2/pull/5652#discussion_r77145686>:
> 
> > @@ -624,6 +1919,13 @@ static int xtensa_op (RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *buf, int l
> >  		return 1;
> >  
> >  	xtensa_op0_fns[(buf[0] & 0xf)] (anal, op, addr, buf);
> > +
> > +	if (anal->decode) {
> > +		uint8_t copy[XTENSA_MAX_LENGTH];
> > +		memcpy (copy, buf, op->size);
> op->size can't be > XTENSA_MAX_LENGTH, but i'll add a check
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/radare/radare2/pull/5652/files/304309e50ef7df936f52888644c3962ce5a0ee72#r77145686>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA3-lg2DeuQukw4FGJ35g7CZQpyA91Dcks5qlqFcgaJpZM4JwqCS>.
>